ecommerce/frontend/src/components/VendorContent/VendorSplash.js

```diff
@@ -0,0 +1,61 @@
+        description: 'Create the future, Save the planet ',
+        rating: 8.7,
+    }
+    const { title, imageUrl, description, rating } = vendorHeaderDetails
+    const { user } = useAppContext()
+    const isVendorAndOwner = user.type === 'vendor' && vendorId === user.id.toString()
+    return (
+        <div className="vendor-splash">
+            <div className="vendor-image">
+                <img src={imageUrl} className="splash-logo"></img>
+            </div>
+            <div className="vendor-header">
+                <div className="vendor-header-content">
+                    <h1 className="vendor-name">{title}</h1>
+                    <h3 className="vendor-slogan">{description}</h3>
+                    <div
+                        className={`product-header-vendor-rating product-header-vendor-rating__${getVendorRatingLevel({
+                            rating,
+                        })}`}>
+                        {round(rating)}
+                    </div>
+                </div>
+                <div className="vendor-edit-button">
+                    {isVendorAndOwner ? (
+                        <div className="vendor-edit-mode-open">
+                            <p>{editable ? 'Edit Mode' : 'Seeing as Customer'}</p>
+                            <Switch defaultChecked={true} onChange={onEditModeChange} />
+                        </div>
```

Comment:
Since you use `onChange` shouldnt you also assign `value={editable}`?

Reply:
Switch component handles its own state.